src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -376,6 +376,18 @@ public function initialized($id)
     }
 
     /**
+     * Clears shared services from the container.
+     */
+    public function clear()
+    {
+        if (!empty($this->scopedServices)) {
+            throw new RuntimeException('Clearing the container is not allowed when a scope is active.');
```

Comment:
note that I added this check because if you are calling ``clear()`` while a scope is active, we can be totally sure that it is a misuse of the method.